### PR TITLE
Add initial implementation

### PR DESCRIPTION
**Pull request by TravisEz13**
cc @tylerleonhardt

### CODE REVIEW

File: src/MacOsPackage.psm1 [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+{
+    param(
+        [parameter(Mandatory)]
+        [string] $Filter,
+        [switch] $IncludeVersion
+    )
+    $results = Start-NativeExecution -ScriptBlock {
+            brew search $Filter  2> $null
+        } -IgnoreExitcode
+    $type=$null
```
Comment (TylerLeonhardt) [inline src/MacOsPackage.psm1:17]:
```suggestion
    $type = $null
```

File: src/MacOsPackage.psm1 [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+    $results | ForEach-Object {
+        if($_ -like '==>*')
+        {
+            switch ($_)
+            {
+                '==> Casks'{
+                    $type=$Script:CaskName
```
Comment (TylerLeonhardt) [inline src/MacOsPackage.psm1:24]:
```suggestion
                    $type = $Script:CaskName
```

File: src/MacOsPackage.psm1 [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+                }
+                '==> Formulae'{
+                    $type=$Script:FormulaName
```
Comment (TylerLeonhardt) [inline src/MacOsPackage.psm1:27]:
```suggestion
                    $type = $Script:FormulaName
```

File: src/MacOsPackage.psm1 [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+function Get-MacOsPackage
+{
+    Get-MacOsPackageFormulae
+    Get-MacOsPackageCask
+}
+
+function Get-MacOsPackageFormulae
+{
+    brew list --full-name --versions -1  2> $null | ForEach-Object {
+        $name,$version=$_ -split ' '
```
Comment (TylerLeonhardt) [inline src/MacOsPackage.psm1:56]:
```suggestion
        $name,$version = $_ -split ' '
```

File: src/MacOsPackage.psm1 [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+            Version = $version
+            Type = $Script:FormulaName
+        }
+    }
+}
+
+function Get-MacOsPackageCask
+{
+    brew cask list -1  2> $null | ForEach-Object {
+        $name=$_
```
Comment (TylerLeonhardt) [inline src/MacOsPackage.psm1:68]:
```suggestion
        $name = $_
```

File: src/MacOsPackage.psm1 [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+
+Function Get-MacOsFormulaeVersion
+{
+    param(
+        [string]$Name
+    )
+
+    $info = brew info $name 2> $null
+    $null,$details = $info[0] -split ': '
+    $channel,$version,$type=$details -split '[ ,\(\)\[\]]{1,3}'
```
Comment (TylerLeonhardt) [inline src/MacOsPackage.psm1:121]:
```suggestion
    $channel,$version,$type = $details -split '[ ,\(\)\[\]]{1,3}'
```

File: src/OsPackage.psd1 [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+Author = 'Travis Plunk'
+
+# Company or vendor of this module
+CompanyName = 'Travis Plunk'
+
+# Copyright statement for this module
+Copyright = '(c) Travis Plunk. All rights reserved.'
+
+# Description of the functionality provided by this module
+# Description = ''
```
Comment (TylerLeonhardt) [inline src/OsPackage.psd1:33]: Add something here

File: src/OsPackage.psd1 [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+@{
+
+# Script module or binary module file associated with this manifest.
+RootModule = 'OsPackage.psm1'
+
+# Version number of this module.
+ModuleVersion = '0.0.1'
+
+# Supported PSEditions
+# CompatiblePSEditions = @()
```
Comment (TylerLeonhardt) [inline src/OsPackage.psd1:18]: Can add `Core` if you'd like

File: src/OsPackage.psd1 [hunk 9]
```diff
@@ -0,0 +1,135 @@
...
+CompanyName = 'Travis Plunk'
+
+# Copyright statement for this module
+Copyright = '(c) Travis Plunk. All rights reserved.'
+
+# Description of the functionality provided by this module
+# Description = ''
+
+# Minimum version of the PowerShell engine required by this module
+# PowerShellVersion = ''
```
Comment (TylerLeonhardt) [inline src/OsPackage.psd1:36]: 6?

File: src/OsPackage.psd1 [hunk 10]
```diff
@@ -0,0 +1,135 @@
...
+# List of all files packaged with this module
+# FileList = @()
+
+# Private data to pass to the module specified in RootModule/ModuleToProcess. This may also contain a PSData hashtable with additional module metadata used by PowerShell.
+PrivateData = @{
+
+    PSData = @{
+
+        # Tags applied to this module. These help with module discovery in online galleries.
+        # Tags = @()
```
Comment (TylerLeonhardt) [inline src/OsPackage.psd1:101]: We should fill out the PSData as well.

**Review by TylerLeonhardt — APPROVED** [review]
LGTM just some whitespace nits and an ask to fill in the module manifest more (also because PowerShellGet requires certain fields to be present

**Comment by TravisEz13** [comment]
I’ll address these in a new PR

**Pull request merged**